### PR TITLE
STS Tests Files and modification in __init__.py

### PR DESCRIPTION
**Pull request by TRYTOBE8TME**
Signed-off-by: Kalpesh Pandya <kapandya@redhat.com>

### CODE REVIEW

File: s3tests_boto3/functional/test_sts.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+@attr(resource='get session token')
+@attr(method='get')
+@attr(operation='check')
+@attr(assertion='s3 ops only accessible by temporary credentials')
+def test_get_session_token():
+    response=""
+    resp=""
+    resp_err=None
+    s3bucket=""
```
Comment (alimaredia) [inline s3tests_boto3/functional/test_sts.py:87]: @TRYTOBE8TME I don't think these variables need to be initialized for every test, I don't think the s3-tests are doing so

Comment (TRYTOBE8TME) [reply s3tests_boto3/functional/test_sts.py:87]: Okay, I'll change this.

File: s3tests_boto3/functional/test_sts.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        role_response = iam_client.put_user_policy(UserName=username,PolicyName=policyname,PolicyDocument=policy_document)
+    except ClientError as e:
+        role_err = e.response['Code']
+    return (role_err,role_response)
+
+@attr(resource='get session token')
+@attr(method='get')
+@attr(operation='check')
+@attr(assertion='s3 ops only accessible by temporary credentials')
+def test_get_session_token():
```
Comment (alimaredia) [inline s3tests_boto3/functional/test_sts.py:80]: For all of your tests can you add a filter similar to this one: https://github.com/ceph/s3-tests/blob/master/s3tests/functional/test_s3.py#L223 to make sure that your tests don't get run with all of the s3-tests and that your tests only get run with the specified filter. Could you also add a section to the README on these sts tests?

Comment (TRYTOBE8TME) [reply s3tests_boto3/functional/test_sts.py:80]: Yes sure, I'll add the sections and do the required changes in the README.

File: s3tests_boto3/functional/__init__.py [hunk 3]
```diff
@@ -205,6 +207,12 @@ def setup():
     config.tenant_user_id = cfg.get('s3 tenant',"user_id")
     config.tenant_email = cfg.get('s3 tenant',"email")
 
+    config.iam_access_key = cfg.get('iam',"access_key")
+    config.iam_secret_key = cfg.get('iam',"secret_key")
+    config.iam_display_name = cfg.get('iam',"display_name")
+    config.iam_user_id = cfg.get('iam',"user_id")
+    #config.iam_email = cfg.get('iam',"email")
```
Comment (alimaredia) [inline s3tests_boto3/functional/__init__.py:214]: if this line is commented out, can you remove it?

Comment (TRYTOBE8TME) [reply s3tests_boto3/functional/__init__.py:214]: I'm not using this but if someone wants then they can use it that's the reason why I thought keeping it.

File: s3tests_boto3/functional/test_sts.py [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+    sts_user_id=get_alt_user_id()
+    default_endpoint=get_config_endpoint()
+    role_session_name=get_parameter_name()
+    policy_document = "{\"Version\":\"2012-10-17\",\"Statement\":[{\"Effect\":\"Allow\",\"Principal\":{\"AWS\":[\"arn:aws:iam:::user/"+sts_user_id+"\"]},\"Action\":[\"sts:AssumeRole\"]}]}"
+    (role_error,role_response,general_role_name)=create_role(iam_client,'/',None,policy_document,None,None,None)
+    eq(role_response['Role']['Arn'],'arn:aws:iam:::role/'+general_role_name+'')
+    role_policy = "{\"Version\":\"2012-10-17\",\"Statement\":{\"Effect\":\"Allow\",\"Action\":\"s3:*\",\"Resource\":\"arn:aws:s3:::*\"}}"
+    (role_err,response)=put_role_policy(iam_client,general_role_name,None,role_policy)
+    eq(response['ResponseMetadata']['HTTPStatusCode'],200)
+    resp=sts_client.assume_role(RoleArn=role_response['Role']['Arn'],RoleSessionName=role_session_name,DurationSeconds=900)
```
Comment (alimaredia) [inline s3tests_boto3/functional/test_sts.py:218]: @TRYTOBE8TME @pritha-srivastava why is the time set to 900 seconds (15 minutes)? Could we make that number lower somehow?

Comment (TRYTOBE8TME) [reply s3tests_boto3/functional/test_sts.py:218]: Actually this test checks that the temporary credentials expire after the said time. And 900 secs is the minimum time required. Thus we need to keep this.

Comment (alimaredia) [reply s3tests_boto3/functional/test_sts.py:218]: Just out of curiosity where is that 900 second minimum set?

File: s3tests_boto3/functional/__init__.py (hunk 3 above)
Comment (alimaredia) [reply s3tests_boto3/functional/__init__.py:214]: If there's no use for it right now in any of the tests could you remove it?

File: s3tests_boto3/functional/__init__.py [hunk 5]
```diff
@@ -246,6 +254,32 @@ def get_v2_client():
...
+
+def get_iam_client(client_config=None):
+    if client_config == None:
+        client_config = Config(signature_version='s3v4')
+    
+    client = boto3.client(service_name='iam',
+                        aws_access_key_id=config.iam_access_key,
+                        aws_secret_access_key=config.iam_secret_key,
+                        endpoint_url=config.default_endpoint,
+                        region_name='',
```
Comment (alimaredia) [inline s3tests_boto3/functional/__init__.py:278]: Why is `region_name` set here? It's not set in any of the other get_client functions

File: s3tests_boto3/functional/test_sts.py (hunk 4 above)
Comment (TRYTOBE8TME) [reply s3tests_boto3/functional/test_sts.py:218]: In the following link: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sts.html#STS.Client.assume_role you can see that in Duration Seconds section.

**Comment by TRYTOBE8TME** [comment]
Okay, I'll think over this after a while because in some days I might also raise a PR for the 3rd API tests. So, I guess after this we can properly generalize them into some functions and have these functions in sts_utils.py. So, would this be fine?

**Pull request merged**